apps/andi/lib/andi/services/dataset_retrieval_service.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Andi.Services.DatasetRetrieval do
+  @moduledoc """
+  Interface for retrieving datasets.
+  """
+  import Andi, only: [instance_name: 0]
+
+  def get_all(instance \\ instance_name()) do
```

Comment:
```elixir
def get_all do
  Brook.get_all_values(instance_name(), :dataset)
end
```

Why are we making `instance_name` a macro to be included in function heads as a default that callers always use anyway? It seems unnecessary.

Reply:
This made it easier to isolate the brook instance we're using for testing.  Using the Andi instance_name in all of our tests modified view state across tests.  By allowing to optionally pass in the instance, we keep that state from being littered across tests that care about this view state.

Reply:
I generally agree that a macro is more optimal from a runtime standpoint. I would honestly love it if Brook gave us the instance name for free as a fallback default via `:application.get_application(__MODULE__)`